bindings/muen/time.c

```diff
@@ -0,0 +1,37 @@
+/*
+ * Copyright (c) 2015-2019 Contributors as noted in the AUTHORS file
+ *
+ * This file is part of Solo5, a sandboxed execution environment.
+ *
+ * Permission to use, copy, modify, and/or distribute this software
+ * for any purpose with or without fee is hereby granted, provided
+ * that the above copyright notice and this permission notice appear
+ * in all copies.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS" AND THE AUTHOR DISCLAIMS ALL
+ * WARRANTIES WITH REGARD TO THIS SOFTWARE INCLUDING ALL IMPLIED
+ * WARRANTIES OF MERCHANTABILITY AND FITNESS. IN NO EVENT SHALL THE
+ * AUTHOR BE LIABLE FOR ANY SPECIAL, DIRECT, INDIRECT, OR
+ * CONSEQUENTIAL DAMAGES OR ANY DAMAGES WHATSOEVER RESULTING FROM LOSS
+ * OF USE, DATA OR PROFITS, WHETHER IN AN ACTION OF CONTRACT,
+ * NEGLIGENCE OR OTHER TORTIOUS ACTION, ARISING OUT OF OR IN
+ * CONNECTION WITH THE USE OR PERFORMANCE OF THIS SOFTWARE.
+ */
+
+#include "bindings.h"
+
+void time_init(const struct hvt_boot_info *bi)
+{
+    assert(tscclock_init(bi->cpu_cycle_freq) == 0);
+}
+
+uint64_t solo5_clock_monotonic(void)
+{
+    return tscclock_monotonic();
+}
+
+/* return wall time in nsecs */
+uint64_t solo5_clock_wall(void)
+{
+    return tscclock_monotonic() + tscclock_epochoffset();
+}
```

Comment:
looks like we need to make a similar change for muen

Reply:
Yes, `muen` is the only one when I did not really change the function. It seems a bit more complex to communicate with the host and ask the wall clock - I don't want to introduce a privilege escalation when `muen` should assert some accesses.